### ISSUE TITLE
Locking individual questions

### ISSUE THREAD

**Issue by nathan-at-jisc** [surveyjs/survey-creator#1107]
### Are you requesting a feature, reporting a bug or ask a question?

Question

### What is the current behavior?

Cannot lock individual questions.

### How would you reproduce the current behavior (if this is a bug)?

At Jisc we run a benchmarking service. This allows uesrs to compare results between multiple surveys (so long as the survey is the same).

However, we allow users to have **benchmarked** questions and **non-benchmarked** questions. The non-benchmarked surveys are now comparable, but the banchmarked ones are. We need to be able to **lock** these types of questions. I know there is a way to do this globally, but is there a way to do this for particular questions?

**Comment by andrewtelnov** [comment]
@nathan-at-jisc You can use [creator.onGetPropertyReadOnly](https://surveyjs.io/Documentation/Survey-Creator?id=surveycreator#onGetPropertyReadOnly) and [creator.onElementAllowOperations](https://surveyjs.io/Documentation/Survey-Creator?id=surveycreator#onElementAllowOperations) events.

Thank you,
Andrew

**Comment by nathan-at-jisc** [comment]
OK nice one.

This works great when creating the survey.

However, we need to save the state that it can't be deleted. Currently I've got:

```json
{
 "pages": [
  {
   "name": "page1",
   "elements": [
    {
     "type": "text",
     "name": "question2"
    },
    {
     "type": "text",
     "name": "question1"
    }
   ]
  }
 ]
}
```

This means that when I come  back to the survey (load it from the DB), it doesn't remember/know that question cannot be deleted/edited. How do I save this state for questions where:

```js
surveyCreator.onElementAllowOperations.add((editor, questionOptions) => {
    questionOptions.allowDelete = false;
});
```
For some background, what we're tryign to achieve is an admin creates a survey and those questions are locked. When this survey is shared (to edit), users *cannot* edit those locked ones, effectively we need something like:

```json
{
 "pages": [
  {
   "name": "page1",
   "elements": [
    {
     "type": "text",
     "name": "question2",
     "locked": true
    },
    {
     "type": "text",
     "name": "question1",
     "locked": true
    }
   ]
  }
 ]
}
```

This way, when a normal user (survey editor), comes to edit a benchmarked survey, some of it is closed off to them. If they attempt to override the JSON, we will detach a chage when uploading to the API and will not allow them to.

**Comment by nathan-at-jisc** [comment]
What would actually be better for us is a `locked` switch on the UI. We can then implement our own logic on who can disable this etc.

**Comment by nathan-at-jisc** [comment]
I think I've answered my own question. I can use: https://surveyjs.io/Examples/Survey-Creator/?id=addproperties#content-js to create a custom option. Then I can use this value to determine whether the value is deletable etc... Does that sound sensible?

**Comment by andrewtelnov** [comment]
@nathan-at-jisc Yes, that is the way to go. You can enable this property for your admin users only.

Thank you,
Andrew
